multiboot2-header/src/header.rs

```diff
@@ -59,6 +61,28 @@ impl<'a> Multiboot2Header<'a> {
         Self { inner: reference }
     }
 
+    /// Find the header in a given slice.
+    pub fn find_header(buffer: &[u8]) -> Option<(&[u8], u32)> {
```

Comment:
> There should be an assertion for correct alignment

I'm not exactly sure, how? The chunking skips 4 bytes each time, so unaligned headers just won't be found.

Reply:
Ah yes, it might be nicer to return an error instead of just `None` in the case of a misaligned header.

With that in mind: Wouldn't it also be nice if `Multiboot2Header::from_addr` returned a `Result` instead of panicking?

Reply:
ah yes, right. Probably it is the best option to add the `Result` type to <https://github.com/rust-osdev/multiboot2/blob/main/multiboot2-header/src/header.rs#L38> and then re-use that error type in the `find_header` function, aye?

I'm thinking of a new `multiboot2_header::LoadError` enum with the following variants:
- invalid address (must be 8-byte aligned and not null)
- MagicNotfound
- ChecksumMismatch